lib/trino-errorprone/src/main/java/io/trino/errorprone/RequireNonNullMessage.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.errorprone;
+
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.MethodInvocationTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.LiteralTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.Tree;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import static com.google.errorprone.matchers.method.MethodMatchers.staticMethod;
+import static com.google.errorprone.util.ASTHelpers.getSymbol;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+@BugPattern(
+        name = "RequireNonNullMessage",
+        summary = "An error message provided to Objects#requireNonNull is incorrect",
+        severity = BugPattern.SeverityLevel.WARNING)
+public class RequireNonNullMessage
+        extends BugChecker
+        implements MethodInvocationTreeMatcher
+{
+    private static final Pattern MESSAGE_PATTERN = Pattern.compile(" (?:(?:is|are|was|were) (?:null|empty|missing|none)|(?:is|are) required|(?:must not|cannot|can't) be (?:null|empty))");
+    private static final Pattern IDENTIFIER_PATTERN = Pattern.compile("^\\p{javaUnicodeIdentifierStart}\\p{javaUnicodeIdentifierPart}*\\b");
+
+    private static final Matcher<ExpressionTree> requireNonNull = staticMethod()
+            .onClass(Objects.class.getName())
+            .named("requireNonNull");
+
+    @Override
+    public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state)
+    {
+        if (!requireNonNull.matches(tree, state)) {
+            return Description.NO_MATCH;
+        }
+        List<? extends ExpressionTree> arguments = tree.getArguments();
+        if (arguments.isEmpty()) {
+            // weird
```

Comment:
why `weird`?

Reply:
There's no no-parameter overload of `Objects#requireNonNull`, but I felt like I should handle that case anyway. Maybe this should be an assertion instead?

Reply:
The code should not fail when JDK gains new methods.